components/prism-graphql.js

```diff
@@ -27,17 +37,50 @@ Prism.languages.graphql = {
 		pattern: /[a-z_]\w*(?=\s*(?:\((?:[^()"]|"(?:\\.|[^\\"\r\n])*")*\))?:)/i,
 		greedy: true
 	},
+	'constant': /\b(?!ID\b)[A-Z][A-Z_\d]*\b/,
 	'class-name': {
-		pattern: /(\b(?:enum|implements|interface|on|scalar|type|union)\s+|&\s*)[a-zA-Z_]\w*/,
-		lookbehind: true
+		pattern: /(\b(?:enum|implements|interface|on|scalar|type|union)\s+|&\s*|:\s*|\s*\[)[A-Z_]\w*/,
+		lookbehind: true,
+		greedy: true
 	},
 	'fragment': {
 		pattern: /(\bfragment\s+|\.{3}\s*(?!on\b))[a-zA-Z_]\w*/,
 		lookbehind: true,
 		alias: 'function'
 	},
-	'keyword': /\b(?:directive|enum|extend|fragment|implements|input|interface|mutation|on|query|repeatable|scalar|schema|subscription|type|union)\b/,
+	'property-mutation': {
+		pattern: /\b(mutation.*(?:[\n\r\u2028\u2029][\n\s]*)?)\b\w+/g,
+		lookbehind: true,
+		greedy: true
+	},
+	'definition-mutation': {
+		pattern: /(\bmutation\s+|\.{3}\s*)[a-zA-Z_]\w*/,
+		lookbehind: true,
+		alias: 'function'
+	},
+	'definition-query': {
+		pattern: /(\bquery\s+|\.{3}\s*)[a-zA-Z_]\w*/,
+		lookbehind: true,
+		alias: 'function'
+	},
+	'keyword-mutation': {
+		pattern: /\b(?:mutation)\b/,
+		alias: 'keyword'
+	},
+	'keyword-query': {
+		pattern: /\b(?:query)\b/,
+		alias: 'keyword'
+	},
+	'keyword': /\b(?:directive|enum|extend|fragment|implements|input|interface|on|repeatable|scalar|schema|subscription|type|union)\b/,
 	'operator': /[!=|&]|\.{3}/,
+	'property-query': {
+		pattern: /\w+(\s*\()/,
+		lookahead: true
+	},
+	'object': {
+		pattern: /\w+(\s*{)/,
+		lookahead: true
+	},
```

Comment:
Why do you need a capture group at all...?  Look-ahead doesn't have to expand the match.

```suggestion
	'object': {
		pattern: /\w+(?=\s*{)/
	},
```